rbi/core/file.rbi

```diff
@@ -662,6 +662,40 @@ class File < IO
   end
   def self.mtime(file); end
 
+  # With no associated block, `File.open` is a synonym for
+  # [`::new`](https://ruby-doc.org/core-2.1.4/File.html#method-c-new).
+  # If the optional code block is given, it will be passed the opened `file` as
+  # an argument and the [`File`](https://ruby-doc.org/core-2.1.4/File.html)
+  # object will automatically be closed when the block terminates.
+  # The value of the block will be returned from `File.open`.
+  #
+  # If a file is being created, its initial permissions may be set using the
+  # `perm` parameter.
+  # See [`::new`](https://ruby-doc.org/core-2.1.4/File.html#method-c-new) for
+  # further discussion.
+  #
+  # See [`IO::new`](https://ruby-doc.org/core-2.6.5/IO.html#method-c-new) for a
+  # description of the `mode` and `opt` parameters.
+  sig do
+    params(
+      filename: String,
+      mode: String,
+      perm: T.nilable(Integer),
+      opt: T.nilable(T::Hash[Symbol, T.untyped]),
+      blk: NilClass
```

Comment:
I think you can also omit `blk` from the overload and it'll work the same way without needing to wait for #2318 to be closed.

Reply:
Sounds good, I removed them and updated the line numbers in the tests. 👍